linq4j/src/main/java/org/apache/calcite/linq4j/tree/Primitive.java

```diff
@@ -385,22 +385,20 @@ static void checkRoundedRange(Number value, double min, double max) {
   }
 
   /** Called from BuiltInMethod.INTEGER_CAST */
-  public static @Nullable Object integerCast(Primitive primitive, final Object value) {
-    return requireNonNull(primitive, "primitive").numberValue((Number) value);
+  public static @Nullable Object integerCast(Primitive primitive, final Object value,
```

Comment:
an alternative is to leave the old API around and just add the new one.
I prefer this solution, but it may cause problems for other projects that try to upgrade to a new version of Calcite.

Reply:
Ok. I will add a new one.